### PR TITLE
Use coursier/interface instead of coursier-small

### PR DESCRIPTION
**Pull request by olafurpg**
Review @poslegm

### CODE REVIEW

File: scalafmt-dynamic/src/main/scala/org/scalafmt/dynamic/ScalafmtDynamicDownloader.scala [hunk 1]
```diff
@@ -4,42 +4,53 @@ import java.io.PrintWriter
...
-        .withRepositories(repositories)
-      val jars: Seq[Path] = CoursierSmall.fetch(settings)
-      val urls = jars.map(_.toUri.toURL).toArray
+      val settings = Fetch
+        .create()
+        .withCache(
+          Cache
+            .create()
+            // TODO: set ttl to infinity, see https://github.com/coursier/interface/issues/57
+            // .withTtl(ttl.orElse(Some(Duration.Inf)))
```
[the remark this answers is not in the document]
Comment (olafurpg) [reply scalafmt-dynamic/src/main/scala/org/scalafmt/dynamic/ScalafmtDynamicDownloader.scala]: That's fine, it won't affect correctness and I think we'll be able to provide the ttl in a later version. I believe the pain of keeping coursier-small is bigger

**Comment by olafurpg** [comment]
Thanks for the review! Let's give this a try 💪

**Comment by olafurpg** [comment]
@tanishiking I took the liberty to publish 2.1.0-RC2. Is there something blocking a 2.1.0 release?

https://github.com/scalameta/scalafmt/releases/tag/v2.1.0-RC2

**Comment by olafurpg** [comment]
Release Drafter is super nice BTW 😄

File: scalafmt-dynamic/src/main/scala/org/scalafmt/dynamic/ScalafmtDynamicDownloader.scala [hunk 2]
```diff
@@ -4,42 +4,53 @@ import java.io.PrintWriter
...
+            .withLogger(Logger.progressBars(downloadProgressWriter))
+        )
+        .withDependencies(dependencies(version).toArray: _*)
+        .withRepositories(repositories: _*)
+        .withCache(Cache.create())
```
[the remark this answers is not in the document]
Comment (poslegm) [reply scalafmt-dynamic/src/main/scala/org/scalafmt/dynamic/ScalafmtDynamicDownloader.scala]: @olafurpg reminder about it. We overwrite cache configuration created at line 27

**Comment by poslegm** [comment]
@olafurpg @tanishiking should I update update scalafmt-dynamic version in sbt-scalafmt now or rather wait for the release 2.1.0?

**Comment by tanishiking** [comment]
Awesome! Thanks, @olafurpg @poslegm 
There are no special blocking issues for 2.1.0. Since we have merged this PR,  I think I'm going to release 2.1.0 soon :)

>  should I update update scalafmt-dynamic version in sbt-scalafmt now or rather wait for the release 2.1.0?

So, I think it's ok to update scalafmt-dynamic to 2.0.1-RC2 now 👍

**Comment by olafurpg** [comment]
We can give RC2 a few days of testing and if everything looks OK we can publish v2.1.0 unchanged :

**Comment by poslegm** [comment]
I pushed 2.0.6 tag in sbt-scalafmt

**Pull request merged**